src/helpers.ml

```diff
@@ -41,21 +41,21 @@ let str_drop_while pred str =
   in inner l
 
 let marshal_int ?(bigendian=true) n x =
-  let s = Bytes.create n in
+  let b = Bytes.create n in
   let f =
     if bigendian
     then function x -> n-x-1
     else function x -> x
   in
   let rec inner num i =
-    if i=n then s else
+    if i=n then b else
       let chr = Int64.logand 0xffL num in
-      Bytes.set s
+      Bytes.set b
         (f i)
         (char_of_int (Int64.to_int chr));
       inner (Int64.shift_right_logical num 8) (i+1)
   in
-  inner x 0
+  inner x 0 |> Bytes.unsafe_to_string
```

Comment:
This is safe IMO because `b` will be lost after the function returns, so nobody can mutate the returned string behind our back.